File: 프로그래머스/배열 원소의 길이.py
# 문자열 배열 strlist가 매개변수로 주어집니다. 
# strlist 각 원소의 길이를 담은 배열을 retrun하도록 solution 함수를 완성해주세요.


# 내 풀이

def solution(strlist):
    answer = []
    for i in strlist:
        answer.append(len(i))
    return answer

# 다른사람 풀이1

def solution(strlist):
    answer = list(map(len, strlist))
    return answer
    
    
# 다른사람 풀이2

def solution(strlist):
    return [len(str) for str in strlist]
    
    
# 다른사람 풀이3

def solution(strlist):
    return list(map(lambda v: len(v), strlist))

File: 프로그래머스/JavaScript/0단계/배열 원소의 길이.js
// strlist 각 원소의 길이를 담은 배열을 retrun하도록 solution 함수를 완성해주세요.

// 내 풀이
function solution(strlist) {
    var answer = [];
    for(let a of strlist){
        answer.push(a.length)
    }
    return answer;
}


// 다른사람 풀이
function solution(strlist) {
    return strlist.map((el) => el.length)  // 배열 돌면서 특정행위를 할 때는 map사용
}
